src/go/rpk/pkg/cli/cmd/redpanda/start.go

```diff
@@ -142,6 +184,15 @@ func NewStartCommand(
 			UnknownFlags: true,
 		},
 		RunE: func(ccmd *cobra.Command, args []string) error {
+			// Since unknown flags are allowed, cobra won't complain.
+			// This might be confusing to the user if they unknowingly make
+			// a typo in one of the flags, so we have to check the flags
+			// ourselves.
+			err := checkFlags(ccmd, []string{setConfigFlag}, append(os.Args, args...))
```

Comment:
I don't think it is necessary to append `args` to `os.Args`, because `os.Args` should already encompass `args`. cobra does not really make it clear what `args` is here, though, but it shouldn't be returning any extra arguments that are not already a part of `os.Args`

Reply:
Actually I think this is capturing the behavior that exists in the tests and only in the tests: `SetArgs` is manually called. 

Will this duplicate the arguments logged in the error message? Since args is `os.Args[1:]`,
and that is appended to `os.Args` here when running rpk not in tests,
then the `checkFlags` command will doubly log each unknown argument,
because `unknown` in that function is a slice?

Reply:
You're right, there's no need to do this. Let me send another version.